### PR TITLE
fix for emojis

### PR DESCRIPTION
**Pull request by ekmb**
Signed-off-by: ekmb <ebakhturina@nvidia.com>

### CODE REVIEW

File: nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py [hunk 1]
```diff
@@ -59,6 +59,11 @@ def get_features(
 
         for j, word in enumerate(words):
             word_tokens = tokenizer.text_to_tokens(word)
+
+            # to handle emojis that could be neglected during tokenization
+            if len(word.strip()) > 0 and len(word_tokens) == 0:
+                word_tokens = [tokenizer.ids_to_tokens(tokenizer.unk_id)]
```
Comment (titu1994) [inline nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py:65]: Does this work for say consecutive multiple emojis (without space or something to seperate as "word")

File: nemo/collections/nlp/data/token_classification/token_classification_dataset.py [hunk 2]
```diff
@@ -87,6 +87,11 @@ def get_features(
 
         for j, word in enumerate(words):
             word_tokens = tokenizer.text_to_tokens(word)
+
+            # to handle emojis that could be neglected during tokenization
+            if len(word.strip()) > 0 and len(word_tokens) == 0:
+                word_tokens = [tokenizer.ids_to_tokens(tokenizer.unk_id)]
```
Comment (titu1994) [inline nemo/collections/nlp/data/token_classification/token_classification_dataset.py:93]: Same as above. Basically we need multiple unk tokens - 1 for every emoji (even if they are consecutive)

File: nemo/collections/nlp/models/nlp_model.py [hunk 3]
```diff
@@ -40,7 +39,6 @@
 from nemo.core.classes.exportable import Exportable
 from nemo.core.connectors.save_restore_connector import SaveRestoreConnector
 from nemo.utils import AppState, logging
-from nemo.utils.exp_manager import configure_checkpointing
```
Comment (titu1994) [inline nemo/collections/nlp/models/nlp_model.py:43]: Why are these being removed? Not  needed?

File: nemo_text_processing/inverse_text_normalization/ru/taggers/tokenize_and_classify.py [hunk 4]
```diff
@@ -114,7 +114,6 @@ def __init__(self, cache_dir: str = None, overwrite_cache: bool = False):
 
             graph = delete_space + graph + delete_space
             self.fst = graph.optimize()
-            generator_main(far_file, {"tokenize_and_classify": self.fst})
```
Comment (titu1994) [inline nemo_text_processing/inverse_text_normalization/ru/taggers/tokenize_and_classify.py:117]: Intentionally removed?

File: nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py (hunk 1 above)
Comment (ekmb) [reply nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py:65]: to extract each emoji we would need to use a package like [ this ](https://pypi.org/project/emoji/)
I do not think it's general enough to add in the processing given the fact that emojies are missing from HF LM vocabs.

Comment (ekmb) [reply nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py:65]: @carolmanderson any opinion on that?

File: nemo/collections/nlp/data/token_classification/token_classification_dataset.py (hunk 2 above)
Comment (ekmb) [reply nemo/collections/nlp/data/token_classification/token_classification_dataset.py:93]: see the above

File: nemo/collections/nlp/models/nlp_model.py (hunk 3 above)
Comment (ekmb) [reply nemo/collections/nlp/models/nlp_model.py:43]: unused imports

File: nemo_text_processing/inverse_text_normalization/ru/taggers/tokenize_and_classify.py (hunk 4 above)
Comment (ekmb) [reply nemo_text_processing/inverse_text_normalization/ru/taggers/tokenize_and_classify.py:117]: yes this lines is duplicated a few lines below

File: nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py (hunk 1 above)
Comment (titu1994) [reply nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py:65]: Ok so for now, if we dont use that package, a string of consecutive emojis would flatten out to a single unk token. As long as that doesnt hurt the model during training or inference, I am ok with that

Comment (carolmanderson) [reply nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py:65]: I think it's good as written. If we have a "word" consisting of multiple emojis, and it fails to generate any subtokens,  I think it's fine to replace that whole word with UNK. If users want to split up runs of multiple emojis, they can do that as a preprocessing step.

**Review by titu1994 — DISMISSED** [review]

File: nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py (hunk 1 above)
Comment (titu1994) [reply nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py:65]: Sounds good. Approved

File: nemo/collections/nlp/modules/common/lm_utils.py [hunk 5]
```diff
@@ -58,6 +58,7 @@ def get_lm_model(
     config_file: Optional[str] = None,
     checkpoint_file: Optional[str] = None,
     vocab_file: Optional[str] = None,
+    model_is_being_restored: Optional[bool] = False,
```
Comment (ericharper) [inline nemo/collections/nlp/modules/common/lm_utils.py:61]: I don't think we need this. `AppState` has the restore flag and you check for it there.

**Review by ericharper — CHANGES_REQUESTED** [review]
Can you try to use `AppState` instead of passing an arg?

File: nemo/collections/nlp/modules/common/lm_utils.py (hunk 5 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/lm_utils.py:61]: https://github.com/NVIDIA/NeMo/blob/1d262c725df150ba837146e7899d8167adf524f6/nemo/utils/app_state.py#L381

Comment (titu1994) [reply nemo/collections/nlp/modules/common/lm_utils.py:61]: Agreed, I'd rather not pass this flag in every single method call, appstate is better for this inside the actual method in one place only.

**Review by titu1994 — CHANGES_REQUESTED** [review]

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Review by titu1994 — APPROVED** [review]
LGTM

**Pull request merged**